mailbox/postgres/src/main/java/org/apache/james/mailbox/postgres/DeleteMessageListener.java

```diff
@@ -146,4 +153,8 @@ private Mono<Boolean> isUnreferenced(PostgresMessageId id, PostgresMailboxMessag
             .map(count -> true)
             .defaultIfEmpty(false);
     }
+
+    private Mono<Void> deleteAttachment(PostgresMessageId messageId, PostgresAttachmentDAO attachmentDAO) {
+        return attachmentDAO.deleteByMessageId(messageId);
```

Comment:
delete blob as well

Reply:
I think we did it in BlobReferenceResource?

Reply:
As I understand, BlobReferenceResource is just to delete duplicated blobs

Reply:
Not really
https://github.com/apache/james-project/blob/fb2fe7da74c784240d5b50963a773465277464c5/server/blob/blob-storage-strategy/src/main/java/org/apache/james/server/blob/deduplication/BloomFilterGCAlgorithm.java#L274

The BlobGC will 
- List all blobId by `blobStoreDAO.listBlobs`
- Check the blobId is referenced anywhere, if not, delete it

Reply:
I think GC is a backup method for the case blob is not deleted in DeleteMessageListener. We still should delete blobs immediately

Reply:
IMO. It is not clear "backup" or not. In another view, the listener is async, it is not "immediately", I can also question for similar cases why we don't delete instantly every reference for every delete event? Why do we tend to have cleanUp tasks in Task Manager?

It 50-50. I will add delete blob as you want, anyway it's simple

Reply:
> I think GC is a backup method for the case blob is not deleted in DeleteMessageListener. We still should delete blobs immediately

Are you sure we should do that? Any chance the attachment is still being referenced somewhere else?

e.g. DeduplicationBlobStore, where the same attachments would share the same object in the blob store.

Reply:
I am not 100% sure though, just rasing the concern

Reply:
Nevermind... `DeDuplicationBlobStore` does not actually delete the blob.

in Cassandra's `DeleteMessageListener`: https://github.com/apache/james-project/blob/b0211356fdc2042db670db0556845419b5bdbaa1/mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/DeleteMessageListener.java#L238

Likely we should delete attachment here.

Reply:
> e.g. DeduplicationBlobStore, where the same attachments would share the same object in the blob store.

DeduplicationBlobStore.deleteBlob always return false, and it do nothing. 

```scala
  override def delete(bucketName: BucketName, blobId: BlobId): Publisher[java.lang.Boolean] = {
    Preconditions.checkNotNull(bucketName)
    Preconditions.checkNotNull(blobId)

    SMono.just(Boolean.box(false))
  }
  ```
  
The trying to delete blob when using  DeduplicationBlobStore doesn't make sense. But yes if PassThroughBlobStore

Reply:
I saw Cassandra DeleteMessageListener delete attachment blob

Reply:
Because Cassandra DeleteMessageListener was coded in 2020
While BlobGC was coded in 2021
haha, just for fun

Reply:
added delete attachmentBlob in https://github.com/apache/james-project/pull/1925/commits/2973dd661dd3d968b811b8e5619ed4882ac2b356